src/languages/php.js

```diff
@@ -431,10 +433,57 @@ export default function(hljs) {
   };
   PARAMS_MODE.contains.push(FUNCTION_INVOKE);
 
+  const ATTRIBUTES = {
+    begin: regex.concat(/#\[\s*/, PASCAL_CASE_CLASS_NAME_RE),
+    beginScope: "meta",
+    end: /]/,
+    endScope: "meta",
+    keywords: {
+      literal: LITERALS,
+      keyword: [
+        'new',
+        'array',
+      ]
+    },
+    contains: [
+      {
+        begin: /\[/,
+        end: /]/,
+        keywords: {
+          literal: LITERALS,
+          keyword: [
+            'new',
+            'array',
+          ]
+        },
+        contains: [
+          'self',
+          NAMED_ARGUMENT,
+          LEFT_AND_RIGHT_SIDE_OF_DOUBLE_COLON,
+          hljs.C_BLOCK_COMMENT_MODE,
+          STRING,
+          NUMBER,
+          CONSTRUCTOR_CALL,
+        ]
+      },
+      NAMED_ARGUMENT,
+      LEFT_AND_RIGHT_SIDE_OF_DOUBLE_COLON,
+      hljs.C_BLOCK_COMMENT_MODE,
+      STRING,
+      NUMBER,
+      CONSTRUCTOR_CALL,
+      {
+        scope: 'meta',
+        match: PASCAL_CASE_CLASS_NAME_RE
```

Comment:
Is it on purpose that this last rule isn't deeply nested like the other contains rules? (line 461-466 vs 469-474)?

Reply:
> Is it on purpose that this last rule isn't deeply nested like the other contains rules? (line 461-466 vs 469-474)?

It is fix for the [second MyAttribute](https://github.com/highlightjs/highlight.js/pull/3407/files#diff-b6b04ba70bc0b19c207aeef76a2e31596393452dba4c39e2c78833d89b63c070R13). Without it, only the first attribute name was highlighted with the meta.

Reply:
Makes sense - and that can never be deeply nested? It has to be top-level?

Reply:
> Makes sense - and that can never be deeply nested? It has to be top-level?

Yes, attributes are only top-level. They can only contain nested classes. Example how IDE highlight them:

![image](https://user-images.githubusercontent.com/57155526/153278884-9d6a34b4-9c56-47a1-85e4-4e0a0fbffc90.png)